pkg/api/types.go

```diff
@@ -978,6 +982,45 @@ type AzureDiskVolumeSource struct {
 	ReadOnly *bool
 }
 
+// ScaleIOVolumeSource represents a persistent ScaleIO volume
+type ScaleIOVolumeSource struct {
+	// The host address of the ScaleIO API Gateway.
+	Gateway string `json:"gateway"`
+	// The name of the storage system as configured in ScaleIO.
+	System string `json:"system"`
+	// SecretRef references to the secret for ScaleIO user and other
+	// sensitive information. If this is not provided, Login operation will fail.
+	SecretRef *LocalObjectReference `json:"secretRef"`
+	// Flag to enable/disable SSL communication with Gateway, default false
+	// +optional
+	SSLEnabled bool `json:"sslEnabled,omitempty"`
+	// The name of the Protection Domain for the configured storage (defaults to "default").
+	// +optional
+	ProtectionDomain string `json:"protectionDomain,omitempty"`
```

Comment:
make them strong typed, refer to https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L470